### ISSUE TITLE
SSH connection to db fails: Can't initialize tunnel - Index 198 out of bounds for length 198 (since update 21.3.5)

### ISSUE THREAD

**Issue by amadeoterri** [dbeaver/dbeaver#15708]
Problems with connecting to database through SSH. I get the following error:

Can't initialize tunnel
  Cannot establish tunnel to {database}
  Cannot establish tunnel to {database}
    fromBase64: invalid base64 data
    fromBase64: invalid base64 data
      Index 198 out of bounds for length 198
      Index 198 out of bounds for length 198

I solved this temporarily by downgrading too 21.3.4..

**Comment by Matvey16** [comment]
@amadeoterri Hello. Thank you for your report. Please provide some details about your configuration. What DB do you use, what OS is DBeaver installed on, if it is possible provide a screenshot with connection settings. And most importantly attach the error log so we can look at the stack trace (https://github.com/dbeaver/dbeaver/wiki/Log-files)

**Comment by ShadelessFox** [comment]
Steps to reproduce:

1. Open `~/.ssh/known_hosts`
2. Remove the last digest from the host's fingerprint base64 string
3. Save
4. Try to connect to this remote host

**Comment by ShadelessFox** [comment]
Hello @amadeoterri,

It looks like your `~/.ssh/known_hosts` file got corrupted.

Can you try erasing it or removing the line containing the host you are trying connecting to?

**Comment by amadeoterri** [comment]
> @amadeoterri Hello. Thank you for your report. Please provide some details about your configuration. What DB do you use, what OS is DBeaver installed on, if it is possible provide a screenshot with connection settings. And most importantly attach the error log so we can look at the stack trace (https://github.com/dbeaver/dbeaver/wiki/Log-files)

I'm using Windows 10. The problem is with plug-in: org.jkis.dbeaver.model. The SSH user name from the screenshots is coupled with my email and public key on the server. I'm using Putty and Pageant to load my keys.
I have the following stacktrace:
```
java.lang.ArrayIndexOutOfBoundsException: Index 198 out of bounds for length 198
	at com.jcraft.jsch.Util.fromBase64(Util.java:52)
	at com.jcraft.jsch.KnownHosts.setKnownHosts(KnownHosts.java:219)
	at com.jcraft.jsch.KnownHosts.setKnownHosts(KnownHosts.java:55)
	at com.jcraft.jsch.JSch.setKnownHosts(JSch.java:316)
	at org.jkiss.dbeaver.model.net.ssh.SSHImplementationJsch.setupTunnel(SSHImplementationJsch.java:110)
	at org.jkiss.dbeaver.model.net.ssh.SSHImplementationAbstract.initTunnel(SSHImplementationAbstract.java:134)
	at org.jkiss.dbeaver.model.net.ssh.SSHTunnelImpl.initializeHandler(SSHTunnelImpl.java:76)
	at org.jkiss.dbeaver.registry.DataSourceDescriptor.connect(DataSourceDescriptor.java:892)
	at org.jkiss.dbeaver.runtime.jobs.ConnectJob.run(ConnectJob.java:70)
	at org.jkiss.dbeaver.runtime.jobs.ConnectJob.runSync(ConnectJob.java:98)
	at org.jkiss.dbeaver.ui.actions.datasource.DataSourceHandler.connectToDataSource(DataSourceHandler.java:113)
	at org.jkiss.dbeaver.ui.actions.datasource.UIServiceConnectionsImpl.initConnection(UIServiceConnectionsImpl.java:70)
	at org.jkiss.dbeaver.model.navigator.DBNDataSource.initializeNode(DBNDataSource.java:158)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.getChildren(DBNDatabaseNode.java:225)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.getChildren(DBNDatabaseNode.java:1)
	at org.jkiss.dbeaver.model.navigator.DBNUtils.getNodeChildrenFiltered(DBNUtils.java:79)
	at org.jkiss.dbeaver.ui.navigator.database.load.TreeLoadService.evaluate(TreeLoadService.java:49)
	at org.jkiss.dbeaver.ui.navigator.database.load.TreeLoadService.evaluate(TreeLoadService.java:1)
	at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:88)
	at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:72)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)

```
![ss1](https://user-images.githubusercontent.com/55751205/156750011-f848cff7-fd14-458f-a5fd-8b7fb07a3a4c.jpg)

![ss2](https://user-images.githubusercontent.com/55751205/156749974-838a592a-d6c8-44a0-bdb0-94e9958d3955.jpg)

![ss3](https://user-images.githubusercontent.com/55751205/156751343-fa08e066-9ea3-4308-a5ec-792666af7ef0.jpg)

**Comment by amadeoterri** [comment]
> Hello @amadeoterri,
> 
> It looks like your `~/.ssh/known_hosts` file got corrupted.
> 
> Can you try erasing it or removing the line containing the host you are trying connecting to?

Ok I will try and let you know. Thanks!

EDIT: But then again, why would my known_hosts be corrupt and still work with DBeaver 21.3.4?

**Comment by ShadelessFox** [comment]
It worked before 21.3.5 because the `~/.ssh/known_hosts` file wasn't used for host key verification, at least in the case of JSch implementation.

**Comment by Matvey16** [comment]
Verified

**Comment by E1izabeth** [comment]
The checkbox to skip host fingerprint verification will be available since version 22.0.0 in Connection settings -> SSH -> Advanced settings.
![image](https://user-images.githubusercontent.com/28875055/156831860-546ed23b-ee84-4598-b9de-dc9371ec9031.png)

**Comment by DariaMarkaryan** [comment]
Verified
